mer-android-chroot

```diff
@@ -298,40 +301,33 @@ do_it_all() {
     run_user_hook mount_sdk
     run_user_hook enter_sdk
 
-    case "$#" in
-	0 )
-	    inform "Entering chroot as $user"
-	    if grep -q squeeze ${uburoot}/etc/debian_version; then
-		# For older 10.04
-		setarch x86_64 chroot ${uburoot} /usr/bin/sudo -i -u $user "export MERSDKUBU=1; if [ -d \"$cwd\" ]; then cd \"$cwd\"; fi; exec bash --init-file /parentroot/usr/share/ubu-chroot/mer-ubusdk-bash-setup -i";
-	    elif grep -w "wheezy\|jessie\|bullseye\|sid" ${uburoot}/etc/debian_version > /dev/null; then
-		# This is for ubuntu 12.04+ (arg quoting for sudo is different)
-		setarch x86_64 chroot ${uburoot} /usr/bin/sudo -i -u $user exec bash -i -c "export MERSDKUBU=1; if [ -d \"$cwd\" ]; then cd \"$cwd\"; fi; exec bash --init-file /parentroot/usr/share/ubu-chroot/mer-ubusdk-bash-setup"
-	    else
-		echo Unknown ubuntu version
-		exit 1
-	    fi
-	    ;;
-	* )
-	    if [[ $1 == 'exec' ]]; then
-		cat <<EOF
-WARN: sdk 'exec' is deprecated. Just execute SDK commands using:
-  $0 <cmd> <args>
-  
-Executing commands as requested
+
+
+    if grep --quiet \
+	    --invert-match \
+	    --word-regexp "wheezy\|jessie\|bullseye\|sid" \
+	    "${uburoot}"/etc/debian_version ; then
+	# This is for ubuntu 12.04+ (arg quoting for sudo is different)
+	echo Unknown ubuntu version >&2
+	exit 1
+    fi
+    if [ $# -eq 0 ] ; then
+        set -- /usr/bin/env \
+            bash --init-file \
+            /parentroot/usr/share/ubu-chroot/mer-ubusdk-bash-setup -i
+    fi
+
+    setarch x86_64 chroot "${uburoot}" \
+            /usr/bin/env su -s /bin/bash -l $user -- \
+            -c "$(cat <<EOF
+[ -d "$cwd" ] && cd "$cwd"
+export MERSDKUBU=1
+
+exec $(printf "%q " "${@}")
 EOF
```

Comment:
While trying to test this I got `su: Authentication failure` when entering ubu-chroot. Rewriting with sudo did the trick:
```suggestion
    if grep --quiet \
	    --invert-match \
	    --word-regexp "wheezy\|jessie\|bullseye\|sid" \
	    "${uburoot}"/etc/debian_version ; then
	# This is for ubuntu 12.04+ (arg quoting for sudo is different)
	echo Unknown ubuntu version >&2
	exit 1
    fi
    if [ $# -eq 0 ] ; then
        set -- /usr/bin/env \
            bash --init-file \
            /parentroot/usr/share/ubu-chroot/mer-ubusdk-bash-setup -i
    fi

    setarch x86_64 chroot "${uburoot}" \
        /usr/bin/sudo -i -u $user \
            exec bash -c "export MERSDKUBU=1; if [ -d \"$cwd\" ]; then cd \"$cwd\"; fi; exec $(printf "%q " "$@")"
```
The last line is the only main change, but I had to also remove the EOF again to make this suggestion into an applicable patch:)

Reply:
Why do that instead of an heredoc without escaping?

Reply:
> While trying to test this I got `su: Authentication failure` when entering ubu-chroot. Rewriting with sudo did the trick:
> 
> The last line is the only main change, but I had to also remove the EOF again to make this suggestion into an applicable patch:)

pwconv should fix this, I'll look.

Reply:
> Why do that instead of an heredoc without escaping?

That worked only after I split commands with `;`, because it becomes a one-liner when using this mnemonic:
```
setarch x86_64 chroot ${uburoot} \
        /usr/bin/sudo -i -u $user \
                exec bash -c "$(cat <<END
export MERSDKUBU=1;
[[ -d "$cwd" ]] && cd "$cwd";
exec $(printf "%q " "$@")
END
)"
```

Reply:
That is because of the redundant exec you have there, it will spawn another shell before executing bash.

Reply:
Please test gain, should be fixed now.